docs/proposals/20231012-pod-decoration.md

```diff
@@ -55,44 +54,59 @@ spec:
     weight: 10      # In same group, heavier PD has higher priority to inject exclusively.
   template:
     metadata:
+    - patchPolicy: Retain    # Retain | Overwrite | MergePatchJson, default Retain
       labels:
         extra-label: label-value
         existing-label: label-value
-    spec:
-      containers:                         # Update by replace. If there is a container with the same name, overwrite it.
-      - name: sidecar
-        image: nginx:1.23.0
-        injectPolicy: BeforeAppContainer  # Position to inject.
-      appContainers:                      # Update by merge. If a name indicated, then merge to the container with the matched name, otherwise update the one indicated by its policy.
-      - image: nginx: 1.23.0
-        env:
-        - name: SIDECAR_INJECTED
-          value: "true"
-        volumeMounts:
-        - name: foo
-          mountPath: /tmp/foo
-        targetSelectPolicy: AllAppContainers  # Indicates this configuration should be injected into all app containers.
-      volumes:
+      annotations:
+        extra-anno: anno-value
+    initContainers:                   
+    - name: init
+      image: init-docker:0.1.0
+    primaryContainers:               # Overwrite old container
+    - targetPolicy: ByName           # ByName | All | First | Last, default by name
```

Comment:
It would be better to use `All` as the default value. There usually is only one primary container.